### ISSUE TITLE
Split out the Unit tests into a separate project

### ISSUE THREAD

**Issue by codedemonuk** [awslabs/amazon-kinesis-client-net#7]
This would allow the DLL for the  client library to not have a dependency on NSubstitute and NUnit.

This would make packaging the project up as a NuGet package easier (see [issue 3](https://github.com/awslabs/amazon-kinesis-client-net/issues/3))

**Comment by pfifer** [comment]
Thanks for reporting this.  We'll investigate splitting the unit tests out as we update the project.  Unfortunately I don't have an ETA at this time.

**Comment by zdjohn** [comment]
i will create a PR for this after previous one is merged.
in my work, i use xunit for testing as it has better support for dotnet core comparing against nunit, moving forward. 
however, i will raise a new issue once the separation is done.
